reader_concurrency_semaphore.cc

```diff
@@ -1023,7 +1025,7 @@ void reader_concurrency_semaphore::consume(reader_permit::impl& permit, resource
 
 void reader_concurrency_semaphore::signal(const resources& r) noexcept {
     _resources += r;
-    maybe_admit_waiters();
+    maybe_wake_execution_loop();
```

Comment:
Avi suggested to repurpose `maybe_admit_waiters()` to what is `maybe_wake_execution_loop()` in this patch.
I'm think this works too.

Reply:
Yes indeed, but then you need to rename the current `maybe_admit_waiters` used in the `execution_loop` into something like `do_admit_waiters`.

When I was working over the code, initially it was hard to find and understand when the `execution_loop` wakes up. Thus, I thing, calling the function `maybe_wake_execution_loop` is more verbose. But I can change it, if that is unacceptable.

Reply:
It is fine by me.

Reply:
Me2